controllers/pwa.js

```diff
@@ -159,6 +159,11 @@ router.post('/add', (req, res, next) => {
  * Display a PWA.
  */
 router.get('/:pwa', (req, res, next) => {
+  const host = req.get('host');
+  const homeUrl = req.protocol + '://' + host;
+  let referer = req.headers.referer &&
```

Comment:
Maybe `up`/`upurl`/`uplink` would be more semantic. Also, I think this can be `const`.